app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/transfers/TransfersScreenWrapper.kt

```diff
@@ -132,20 +144,71 @@ private fun DetailPane(
     navigator: ThreePaneScaffoldNavigator<DestinationContent>,
     hasTransfer: () -> Boolean,
 ) {
-
     val destinationContent = navigator.safeCurrentContent()
 
     if (destinationContent == null) {
         NoSelectionEmptyState(hasTransfer())
+    } else if (destinationContent.folderUuid != null) {
+        val windowAdaptiveInfo = LocalWindowAdaptiveInfo.current
+        FilesDetailsScreen(
+            windowAdaptiveInfo,
+            navigator,
+            destinationContent.folderUuid,
+            destinationContent.direction,
+            destinationContent.transferUuid,
+            navigateToFolder = { selectedFolderUuid ->
+                navigator.navigateToFolder(
+                    destinationContent.direction,
+                    destinationContent.transferUuid,
+                    selectedFolderUuid,
+                )
+            }
+        )
     } else {
         TransferDetailsScreen(
             transferUuid = destinationContent.transferUuid,
             direction = destinationContent.direction,
             navigateBack = ScreenWrapperUtils.getBackNavigation(navigator),
+            navigateToFolder = { selectedFolderUuid ->
+                navigator.navigateToFolder(
+                    destinationContent.direction,
+                    destinationContent.transferUuid,
+                    selectedFolderUuid,
+                )
+            },
         )
     }
 }
 
+@OptIn(ExperimentalMaterial3AdaptiveApi::class)
+@Composable
+private fun FilesDetailsScreen(
+    windowAdaptiveInfo: WindowAdaptiveInfo,
+    navigator: ThreePaneScaffoldNavigator<DestinationContent>,
+    folderUuid: String,
+    transferDirection: TransferDirection,
+    transferUuid: String,
+    navigateToFolder: (folderUuid: String) -> Unit,
+) {
```

Comment:
Is this the counterpart of the FilesDetailsScreen that uses a view model declared in FilesDetailsScreen.kt?